### PR TITLE
Migrate to setuptools 

### LINKED ISSUE #4: Migrate from distutils to setuptools
Hi Damien,

For a bit of context, I currently use @drprojects/superpoint_transformer to make some forest segmentation for the @3DFin project. It works pretty well but we want to lower the technical cost of using it (in an inference context) for scientist interested in the field of Forest management. 

I started to review compiled dependency and evaluate if we can offer pre build wheels (via https://github.com/pypa/cibuildwheel). I first noticed that all your C++ based python module/dependencies use `distutils` as a build system. `distutils` is deprecated and I think it did not even work anymore with `python 3.12`. So future users could have hard time to install your dependencies. Would you agree if I try to port `pgeof` to `setuptools `build system (which is the "drop in" replacement for `distutils`)?

### PR DESCRIPTION
**Pull request by rjanvier**
Close #4

- Reorganize the file hierarchy. The module live inside the `src` dir. Build files are located in the root dir. This should be mirrored into SPT install script (I will make dedicated a PR. see an example against my branch https://github.com/rjanvier/superpoint_transformer/blob/a07d92db07e1cdf8a228cb6d26814257bf6d7f85/install.sh#L106-L112)
- Migrate to setuptools. Pgeof install script is now pep 517 "compliant". it could be installed by symply using `pip install .` on the root of the directory. In order to achieve this, the way Eigen lib include dir is passed to the build system was changed. We now use env variables because of deprecation of some config flags in pip (https://github.com/pypa/pip/issues/11859) and because setuptools does not implemente alternatives (https://github.com/pypa/setuptools/issues/2491) yet.
- Use cibuildwheels to produce pre compiled binaries for Windows and Linux. The build is triggered on each PR/Push so it could be used a CI system too. The upload on PyPI is triggered when tagging a specific commit. You have to setup a PiPY account and to set `secrets.PYPI_TOKEN` variable on your github repo.

### CODE REVIEW

**Comment by rjanvier** [comment]
Seems you close this without merging it :)

**Review by drprojects — CHANGES_REQUESTED** [review]

File: README.md [hunk 1]
```diff
@@ -30,12 +30,26 @@ Python wrapper around C++ helper to compute, for each point in a 3D point cloud,
 - volume
 - curvature
 - optimal neighborhood size
-
 </details>
 
 
 ## 🧱 Installation
 
+```bash
```
Comment (drprojects) [inline README.md:38]: Could you please at some subsections to clarify the two installation modes ? 

## 🧱 Installation

### Inside an existing environment 

```bash
# clone project
git clone https://github.com/drprojects/point_geometric_features.git
cd point_geometric_features

# run 
python -m pip install .
```

Or, if you do not need the source files:

```bash
python -m pip install git+https://github.com/drprojects/point_geometric_features
```

Pgeof depends on [Eigen library](https://eigen.tuxfamily.org/) and numpy headers at build time.
The good version of Numpy will be fetched from PyPI automatically but your are responsible for providing
the path to the Eigen library you want to use (for example by using `CXXFLAGS` variable on Linux or setting `EIGEN_LIB_PATH`).

Pgeof will be soon available as pre compiled package on PyPI.

### In a new dedicated `conda` environment 

The following will install the project in a new `conda` environment named `pgeof`.

```bash
# clone project
git clone https://github.com/drprojects/point_geometric_features.git
cd point_geometric_features

# Installation in a new dedicated `pgeof` conda environment
bash install.sh
```

You can easily adapt `install.sh` to install the project in an already-existing 
environment.

**Review by drprojects — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -30,12 +30,26 @@ Python wrapper around C++ helper to compute, for each point in a 3D point cloud,
...
+# clone project
+git clone https://github.com/drprojects/point_geometric_features.git
+cd point_geometric_features
+
+# run 
+pyhon -m pip install .
```
[the remark this answers is not in the document]
Comment (rjanvier) [reply README.md:44]: yes, thanks.

File: README.md (hunk 1 above)
Comment (rjanvier) [reply README.md:38]: Ok!

File: README.md [hunk 3]
```diff
@@ -50,13 +74,12 @@ bash install.sh
...
 environment.
 
-
 ## 🚀 Using Point Geometric Features
 
 The `pgeof` function should be used as follows:
 
 ```python
-from src import pgeof
+import pgeof
```
Comment (drprojects) [inline README.md:82]: If I am not mistaken, it is:

```python
from pgeof import pgeof
```

File: demo.py [hunk 4]
```diff
@@ -1,7 +1,4 @@
-import sys
-import os.path as osp
-sys.path.append(osp.join(osp.realpath(osp.dirname(__file__))))
-from src import pgeof
+import pgeof
```
Comment (drprojects) [inline demo.py:1]: Same comment here:

```python
from pgeof import pgeof
```

**Review by drprojects — CHANGES_REQUESTED** [review]

**Review by drprojects — APPROVED** [review]

**Comment by drprojects** [comment]
Thanks for your edits, this seems good to me 👍

**Pull request merged**